src/foremast/awslambda/awslambda.py

```diff
@@ -54,10 +54,12 @@ def __init__(self, app, env, region, prop_path):
         self.handler = self.pipeline['handler']
         self.vpc_enabled = self.pipeline['vpc_enabled']
 
-        self.memory = self.properties[env]['app']['lambda_memory']
-        self.timeout = self.properties[env]['app']['lambda_timeout']
+        app = self.properties[self.env]['app']
+        self.memory = app['lambda_memory']
+        self.role = app.get('lambda_role') or generated.iam()['lambda_role']
```

Comment:
It looks like this happens because we never try to create the newly named role that comes from `generated.iam()['lambda_role']`, we only ever create `generated.iam()['role']`